ext/mbstring/mbstring.c

```diff
@@ -5641,3 +5613,232 @@ static void php_mb_gpc_set_input_encoding(const zend_encoding *encoding) /* {{{
 	MBSTRG(http_input_identify) = (const mbfl_encoding*)encoding;
 }
 /* }}} */
+
+static int8_t decode_base64(unsigned char c)
```

Comment:
can standard base64 decode function be used here? especially with #6361

Reply:
Thanks for the suggestion. I'll have to check what its interface looks like and see if it fits what is needed here.

Reply:
Hmm, the functions provided by `base64.c` take a pointer/length pair (good) and return a `zend_string` (not what I wanted).

But maybe I can adjust the code to use a `zend_string`. Let me think about that a bit.

Reply:
Or perhaps it's possible to make the other code internally work with a pointer/length pair. And let the existing callers of that code work with a wrapper for zend_string that extracts the pointer & length.